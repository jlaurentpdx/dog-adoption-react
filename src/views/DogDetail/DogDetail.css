h1 {
  color: var(--darker);
  font-size: 300%;
  font-family: 'Indie Flower', sans-serif;
}

h2,
h3 {
  color: var(--darker);
  font-family: 'Karla', sans-serif;
  font-weight: 300;
  font-size: 3rem;
}

h3 {
  font-size: 2.5rem;
}

p {
  color: var(--dark);
  font-family: 'Indie Flower', sans-serif;
  font-weight: 200;
  font-size: 1.8rem;
}
