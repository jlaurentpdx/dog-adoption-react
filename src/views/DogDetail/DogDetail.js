import { useEffect, useState } from 'react';
import { fetchDogById } from '../../services/dogs';

export default function DogDetail(props) {
  const id = props.match.params.id;
  const [loading, setLoading] = useState(true);
  const [dog, setDog] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(id);
      console.log(data);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  if (loading) return <h1>Please wait...</h1>;
  return (
    <div className="dog-detail">
      <h1>{dog.name}</h1>
      <img src={dog.image} />
      <p>
        Meet {dog.name}, the {dog.breed}, age {dog.age}.
      </p>
      <p>{dog.bio}</p>
    </div>
  );
}
