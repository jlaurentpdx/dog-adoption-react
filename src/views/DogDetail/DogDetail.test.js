import { render, screen } from '@testing-library/react';
import DogDetail from './DogDetail';

test('DogDetail renders an individual dog with details', async () => {
  const { container } = render(<DogDetail match={{ params: { id: 14 } }} />);

  screen.getByText('...getting ready to introduce...');

  await screen.findByText('Eyes');
  await screen.findByText('Gentle giant with eyes that will melt your soul.');

  expect(container).toMatchSnapshot();
});
