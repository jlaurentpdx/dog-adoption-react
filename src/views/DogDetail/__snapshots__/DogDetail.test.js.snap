// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogDetail renders an individual dog with details 1`] = `
<div>
  <div
    class="dog-detail"
  >
    <h1>
      Sadie
    </h1>
    <img
      src="https://placedog.net/400/400/6"
    />
    <h2>
      Meet 
      Sadie
      , the 
      
    </h2>
    <h3>
      Age 
      10
      .
    </h3>
    <p>
      When she's not busy being your admirer, Sadie likes to curl up and take a nap. Great with kids and other mellow pets!
    </p>
  </div>
  <a
    href="/dogs/6/edit"
  >
    Edit
  </a>
</div>
`;
